pkg/scheduler/framework/cycle_state.go

```diff
@@ -79,15 +79,25 @@ func (c *CycleState) SetRecordPluginMetrics(flag bool) {
 
 // Clone creates a copy of CycleState and returns its pointer. Clone returns
 // nil if the context being cloned is nil.
+//
+// Clone is a special function only used in preemption. When all Nodes are
```

Comment:
This is used both in preemption and normal filtering stage, as you mentioned, you can refer to `RunFilterPluginsWithNominatedPods`'s comment:
```
// This function is called from two different places: Schedule and Preempt.
```

So I think we should revert  this.

Reply:
I think it would be better to add why Schedule is requiring to clone state vs remove docs about Preempt. It took many days to accumulate below. If there's a better place to house this information let me know

Reply:
I'll remove it. I think the answer to my question would probably be as confusing as the notes below, not because of you @kerthcet personally. It is possible some things should just not be documented

Reply:
reverted rationale